### PR TITLE
ARROW-5813: [C++] Fix TensorEquals for different contiguous tensors

### PR DESCRIPTION
**Pull request by mrkn**
This change makes TensorEquals correctly calculate the equality of a row-major tensor and a column-major tensor.

### CODE REVIEW

File: cpp/src/arrow/tensor-test.cc [hunk 1]
```diff
@@ -155,6 +155,49 @@ TEST(TestTensor, CountNonZeroForNonContiguousTensor) {
   AssertCountNonZero(t, 8);
 }
 
+TEST(TestTensor, Equals) {
```
Comment (pitrou) [inline cpp/src/arrow/tensor-test.cc]: Is it possible to also test unequal tensors? Contiguous, non-contiguous...

**Review by pitrou — CHANGES_REQUESTED** [review]
Looks good on the principle, but I think testing should be a bit more comprehensive perhaps?

**Comment by pitrou** [comment]
By the way, `memcmp` is problematic for floating-point data (because of NaNs at least, and possibly denormals - but not sure we care about the latter). That can be tackled in a separate JIRA though.

File: cpp/src/arrow/tensor-test.cc (hunk 1 above)
Comment (mrkn) [reply cpp/src/arrow/tensor-test.cc]: Oh, I forgot to write unequal cases.
I added them at d40c3c7745a318507f05f174ae9dac5a1d381718

**Comment by mrkn** [comment]
I filed [ARROW-5830](https://issues.apache.org/jira/browse/ARROW-5830).

**Review by pitrou — APPROVED** [review]

**Comment by pitrou** [comment]
Thanks @mrkn ! Will merge.

**Pull request closed without merging**